### PR TITLE
Run code checks and tests on pull requests and merges to master

### LINKED ISSUE #13: Add GitLab Actions config to run checks and tests on pull requests and merges into master
Basically the same as localgovdrupal/localgov_services#38

### PR DESCRIPTION
**Pull request by stephen-cox**
Closes #13

### CODE REVIEW

**Comment by stephen-cox** [comment]
@ekes This adds checks and tests on the guides stuff. The tests are failing because they depend on the services feature/31-service-paths-other branch - which highlights the issue with this I raised at stand-up

**Comment by ekes** [comment]
It's running correctly - and hence failing - so it seems good to go.

Which makes me think.

A PR that involves multiple branches maybe should usually be on the project?
You could also alter the branch checked out on the project in the test.yml - more things to keep track of :-(
Or - if this commit that relies on feature/31-service-paths-other was a PR (which it's not at the moment because it was all part of the dev but) then it should fail until feature/31-service-paths-other is committed first.

**Comment by ekes** [comment]
> A PR that involves multiple branches maybe should usually be on the project? [edit: actually usually not - as they're often not things that depend on the project being installed, and can be installed independently -- although it is a good way of testing]

Sorry I'm thinking fractionally slower than I'm typing:

So to test thing module you don't need to install the project. The dependencies for the tests for this module are in the composer.json
So to run the tests for this module you should need to do a _standard_ drupal composer install, and a require on this module.

The composer.json in this module should then (temporarily) include any branch requirements.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    steps:
+      - name: Clone drupal_container
+        uses: actions/checkout@v2
+        with:
+          repository: localgovdrupal/drupal-container
+          ref: master
+
+      - name: Create LocalGovDrupal project
+        run: composer create-project --stability dev localgovdrupal/localgov-project html
```
Comment (ekes) [inline .github/workflows/test.yml:22]: So we can just install stock drupal here.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      - name: Checkout localgov_guides branch to test
+        uses: actions/checkout@v2
+        with:
+          path: html/web/modules/contrib/localgov_guides
```
Comment (ekes) [inline .github/workflows/test.yml:27]: Then would it be possible to composer require the appropriate branch being tested here? It's a variable in the action I think.

**Comment by finnlewis** [comment]
@ekes I think it would be good to merge this in to get basic tests running.

More generally it would be good to be able to require the appropriate branch, where best to raise an issue for that?

**Pull request merged**